pkg/datapath/linux/node.go

```diff
@@ -562,9 +562,9 @@ func (n *linuxNodeHandler) insertNeighbor(newNode *nodeTypes.Node, ifaceName str
 		return
 	}
 
-	_, err = arping.FindIPInNetworkFromIface(ciliumIPv4, *iface)
-	if err != nil {
-		scopedLog.WithError(err).Error("IP is not L2 reachable")
+	hwAddr, pingErr := arp.PingOverIface(*iface, ciliumIPv4)
```

Comment:
I'm not sure off-hand what the old implementation did, but we might consider some retry logic here otherwise we can't add a neighbor and it gets difficult to recover from later.

Reply:
There was no retries around the places I modified. It should not be hard to add it with the implementation from this PR.